### ISSUE TITLE
emit to ostream

### ISSUE THREAD

**Issue by jwuttke** [jbeder/yaml-cpp#337]
Emitting through c_str() is not quite C++ish. At least, there should be a method that returns a C++ string. Preferably, there should be a way to send the YAML::Emitter to any ostream. Ideally, it were possible to connect the YAML::Emitter to an ostream before sending contents to the emitter. Any contents would then appear at once in the output file, which would facilitate debugging in case something goes wrong before everything is sent to the emitter.

**Comment by SGSSGene** [comment]
True, but currently no time to improve it. If anyone feels like submitting a PR we will take a look at it.
